### PR TITLE
doc: note that durations must not be negative

### PR DESCRIPTION
**Pull request by kortschak**
Fixes ev3dev/ev3dev#744.

@dlech Please take a look.

### CODE REVIEW

File: motors/dc_motor_class.c [hunk 1]
```diff
@@ -112,7 +112,7 @@
  * .
  * `time_sp`
  * : (read/write) Sets the time setpoint used with the `run-timed` command.
- *   Units are in milliseconds.
+ *   Units are in milliseconds. Values must not be negative.
```
Comment (dlech) [inline motors/dc_motor_class.c]: does writing a negative number to this attribute return `-EINVAL`? If not, it probably should.

**Comment by dlech** [comment]
I guess it is just intuitive to me that you would never have a negative time value. It doesn't seem necessary to me to explicit state that.

File: motors/dc_motor_class.c (hunk 1 above)
Comment (kortschak) [inline motors/dc_motor_class.c]: I dont have a dc motor lead to test that with.

**Comment by kortschak** [comment]
In a variety of contexts, negative duration values are used to indicate no-limit. These additions deny that here.

File: motors/dc_motor_class.c (hunk 1 above)
Comment (kortschak) [inline motors/dc_motor_class.c]: No worries. In many cases I have checked. I will do an exhaustive check tonight (currently apt upgrading and then going to work).

**Comment by kortschak** [comment]
I've done the checks I can for whether the attributes do indeed baulk at negative values.

In all cases I tried `echo -1 > <attr>`. The checked attributes printed `write error: Invalid argument`.

dc-motor:
- [x] `time_sp`

servo-motor:
- [ ] `time_sp` (could not check cannot set out port mode to servo-motor)

tacho-motor:
- [x] `ramp_up_sp`
- [x] `ramp_down_sp`
- [x] `time_sp`

lego-sensor:
- [ ] `poll_ms` (not able to check - no available device with modifiable poll rate)

nxt-i2c:
- [x] `default_poll_ms` (done as root)

**Comment by dlech** [comment]
> could not check - cannot set out port mode to servo-motor

The servo motor class doesn't have `time_sp` but the other time based setpoints have range checking. [example](https://github.com/ev3dev/lego-linux-drivers/blob/master/motors/servo_motor_class.c#L257).

> not able to check - no available device with modifiable poll rate

'poll_ms` is [parsing unsigned ints](https://github.com/ev3dev/lego-linux-drivers/blob/5ccdd2dc17f097d7a766ecad18b54d5372fbf88c/sensors/lego_sensor_class.c#L515), so we should be good here.

**Comment by dlech** [comment]
So, LGTM. Will you update the commit message with a reference to the issue?

**Comment by kortschak** [comment]
Yes, I'll do that.

**Comment by kortschak** [comment]
>The servo motor class doesn't have `time_sp`

Ugh, I can't even read my own commits.

**Comment by dlech** [comment]
Thanks!

**Pull request merged**